lumen/ai/logs.py

```diff
@@ -1,91 +1,380 @@
 from __future__ import annotations
 
 import sqlite3
+import uuid
 
+from typing import Literal
+
+import pandas as pd
 import param
 
+from panel import Card
+from panel.chat import ChatMessage
+from panel.widgets import Tabulator
+
 from lumen.ai.utils import log_debug
 
 
 class ChatLogs(param.Parameterized):
+    """A class for managing and storing chat message logs in SQLite.
+
+    The class handles storage, retrieval, and versioning of chat messages,
+    including support for message retries and status tracking.
+    """
 
-    filename = param.String(default="chat_logs.db")
+    filename: str = param.String(default="lumen_chat_logs.db")
+    session_id: str = param.String(constant=True)
 
-    def __init__(self, **params):
+    def __init__(self, **params) -> None:
+        if "session_id" not in params:
+            params["session_id"] = uuid.uuid4().hex[:8]
         super().__init__(**params)
-        self.conn = sqlite3.connect(self.filename)
-        self.cursor = self.conn.cursor()
-        self.cursor.execute(
-            """
-            CREATE TABLE IF NOT EXISTS logs (
-                session_id TEXT,
+        self.conn: sqlite3.Connection = sqlite3.connect(self.filename)
+        self.cursor: sqlite3.Cursor = self.conn.cursor()
+
+        # Create messages table
+        self.cursor.execute("""
+            CREATE TABLE IF NOT EXISTS messages (
                 message_id TEXT PRIMARY KEY,
+                session_id TEXT,
                 message_index INTEGER,
                 message_user TEXT,
                 message_content TEXT,
+                attempt_number INTEGER DEFAULT 0,
                 liked BOOLEAN DEFAULT FALSE,
                 disliked BOOLEAN DEFAULT FALSE,
-                removed BOOLEAN DEFAULT FALSE,
+                state TEXT DEFAULT 'active',
```

Comment:
Used state instead of `removed` to have a clearer picture of what was changed to the message: `Literal["active", "reran", "undone", "cleared", "edited", "retried"]`